diff --git a/meta-oe/recipes-connectivity/gnuradio/gnuradio_git.bb b/meta-oe/recipes-connectivity/gnuradio/gnuradio_git.bb
index 1a78575..edd4615 100644
--- a/meta-oe/recipes-connectivity/gnuradio/gnuradio_git.bb
+++ b/meta-oe/recipes-connectivity/gnuradio/gnuradio_git.bb
@@ -27,7 +27,7 @@ RDEPENDS_${PN} = "python-core python-audio python-threading python-codecs \
                   python-netclient python-difflib \
                   python-pprint python-numpy  \
 "
-RDEPENDS_${PN}-grc = "python-pygtk python-lxml python-cheetah python-netserver"
+RDEPENDS_${PN}-grc = "python-pygtk python-lxml python-cheetah"
 
 C_FILES_CHKSUM = "file://COPYING;md5=d32239bcb673463ab874e80d47fae504"
 
@@ -43,35 +43,34 @@ do_compile_prepend() {
 }
 
 PACKAGES += " \
-    ${PN}-grc \
     ${PN}-examples \
+    ${PN}-grc \
 "
 
 FILES_${PN}-grc = "${datadir}/gnuradio/grc"
-FILES_${PN}-examples = "${datadir}/gnuradio/examples"
 
-FILES_${PN} += "${PYTHON_SITEPACKAGES_DIR}/gnuradio"
-FILES_${PN} += "${datadir}/gnuradio/modtool"
+FILES_${PN} += "${PYTHON_SITEPACKAGES_DIR}/gnuradio/*"
+FILES_${PN} += "${datadir}/gnuradio/*"
 # The following needs fixing upstream
-FILES_${PN} += "${prefix}/etc/gnuradio"
+FILES_${PN} += "${prefix}/etc/gnuradio/*"
 
 FILES_${PN}-dbg += "${PYTHON_SITEPACKAGES_DIR}/gnuradio/.debug \
                     ${PYTHON_SITEPACKAGES_DIR}/gnuradio/*/.debug \
                     ${datadir}/gnuradio/examples/*/.debug \
 "
+FILES_${PN}-examples = "${datadir}/gnuradio/examples"
 
-PV = "3.6.5"
+PV = "3.6.4.1"
 
 FILESPATHPKG_prepend = "gnuradio-git:"
 
-SRCREV = "a7b3a9339106aaba72195b693dfeaa98f194e0d5"
+SRCREV = "5f69899e059e9bea58f92af61f70fc3f63825087"
 
 # Make it easy to test against developer repos and branches
 GIT_REPO = "gnuradio.git"
 GIT_BRANCH = "master"
 
-SRC_URI = "git://git.gnuradio.org/${GIT_REPO};branch=${GIT_BRANCH} \
-           file://0001-gr-trellis-Kill-docs-hard.patch \
+SRC_URI = "git://gnuradio.org/${GIT_REPO};branch=${GIT_BRANCH};protocol=git \
 "
 
 S="${WORKDIR}/git"
@@ -83,18 +82,21 @@ EXTRA_OECMAKE = "-DENABLE_GR_ATSC=FALSE \
                  -DENABLE_GR_FCD=OFF \
                  -DENABLE_GR_WXGUI=OFF \
                  -DENABLE_GR_VIDEO_SDL=OFF \
-                 -DENABLE_SPHINX=OFF -DENABLE_DOXYGEN=OFF \
                  -DIMPORT_EXECUTABLES=${S}/gr-vocoder/lib/generate_codebook.txt \
                  -DQT_HEADERS_DIR=${STAGING_INCDIR}/qt4 \
                  -DQT_QTCORE_INCLUDE_DIR=${STAGING_INCDIR}/qt4/QtCore \
                  -DQT_LIBRARY_DIR=${STAGING_LIBDIR} \
                  -DQT_QTCORE_LIBRARY_RELEASE=${STAGING_LIBDIR}/libQtCore.so \
                  -DQT_QTGUI_LIBRARY_RELEASE=${STAGING_LIBDIR}/libQtGui.so \
-                 ${@base_contains('TUNE_FEATURES', 'neon', \
-                     '-Dhave_mfpu_neon=1', '-Dhave_mfpu_neon=0', d)} \
 "
 
 inherit distutils-base cmake pkgconfig
 
 EXTRA_OEMAKE = "-C ${OECMAKE_BUILDPATH}"
 
+# Only builds for machines with neon instructions. Problem is in upstream cmake.
+python () {
+    if not oe.utils.contains ('TUNE_FEATURES', 'neon', True, False, d):
+        raise bb.parse.SkipPackage("'neon' not in TUNE_FEATURES")
+}
+
